op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -88,332 +115,253 @@ var (
+		},
+	}
 
-			if config.DeployCeloContracts {
-				return errors.New("DeployCeloContracts is not supported in migration")
-			}
-			if config.FundDevAccounts {
-				return errors.New("FundDevAccounts is not supported in migration")
-			}
+	if err := app.Run(os.Args); err != nil {
+		log.Crit("error in migration", "err", err)
+	}
+	log.Info("Finished migration successfully!")
+}
 
-			// Try reading the L1 deployment information
-			deployments, err := genesis.NewL1Deployments(l1Deployments)
-			if err != nil {
-				return fmt.Errorf("cannot read L1 deployments at %s: %w", l1Deployments, err)
-			}
-			config.SetDeployments(deployments)
+func runBlockMigration(opts blockMigrationOptions) error {
 
-			// Get latest block information from L1
-			var l1StartBlock *types.Block
-			client, err := ethclient.Dial(l1RPC)
-			if err != nil {
-				return fmt.Errorf("cannot dial %s: %w", l1RPC, err)
-			}
+	// Check that `rsync` command is available. We use this to copy the db excluding ancients, which we will copy separately
+	if _, err := exec.LookPath("rsync"); err != nil {
+		return fmt.Errorf("Please install `rsync` to run block migration")
+	}
 
-			if config.L1StartingBlockTag == nil {
-				l1StartBlock, err = client.BlockByNumber(context.Background(), nil)
-				if err != nil {
-					return fmt.Errorf("cannot fetch latest block: %w", err)
-				}
-				tag := rpc.BlockNumberOrHashWithHash(l1StartBlock.Hash(), true)
-				config.L1StartingBlockTag = (*genesis.MarshalableRPCBlockNumberOrHash)(&tag)
-			} else if config.L1StartingBlockTag.BlockHash != nil {
-				l1StartBlock, err = client.BlockByHash(context.Background(), *config.L1StartingBlockTag.BlockHash)
-				if err != nil {
-					return fmt.Errorf("cannot fetch block by hash: %w", err)
-				}
-			} else if config.L1StartingBlockTag.BlockNumber != nil {
-				l1StartBlock, err = client.BlockByNumber(context.Background(), big.NewInt(config.L1StartingBlockTag.BlockNumber.Int64()))
-				if err != nil {
-					return fmt.Errorf("cannot fetch block by number: %w", err)
-				}
-			}
+	debug.SetMemoryLimit(opts.memoryLimit * 1 << 20) // Set memory limit, converting from MB to bytes
 
-			// Ensure that there is a starting L1 block
-			if l1StartBlock == nil {
-				return fmt.Errorf("no starting L1 block")
-			}
+	log.Info("Block Migration Started", "oldDBPath", opts.oldDBPath, "newDBPath", opts.newDBPath, "batchSize", opts.batchSize, "memoryLimit", opts.memoryLimit, "clearAll", opts.clearAll, "keepNonAncients", opts.keepNonAncients, "onlyAncients", opts.onlyAncients)
 
-			// Sanity check the config. Do this after filling in the L1StartingBlockTag
-			// if it is not defined.
-			if err := config.Check(); err != nil {
-				return err
-			}
+	var err error
+
+	if opts.clearAll {
+		if err = os.RemoveAll(opts.newDBPath); err != nil {
+			return fmt.Errorf("failed to remove new database: %w", err)
+		}
+	} else if !opts.keepNonAncients {
+		if err = cleanupNonAncientDb(opts.newDBPath); err != nil {
```

Comment:
Since this check goes first, it would fail if there is no new db yet as it's created only on [line 248](https://github.com/celo-org/optimism/pull/148/files#diff-e9ab083305d85cb9fea217fc18988c2748f3918b328559125eba8db4e33e6da4R248) in the case when neither `clear-all` nor `keep-nonAncients` flag is set. I believe it should be safe to call `createNewDbIfNotExists` before everything else.